wiki/Lua-Hyperspace-Module.md

```diff
@@ -1229,6 +1797,273 @@ local _, canMove = crew.extend:CalculateStat(Hyperspace.CrewStat.CAN_MOVE)
+- `int` `.replaceGroupIndex`
+   - **read-only**
+- `int` `.sortOrder`
+- [`Damage`](#Damage) `.damage`
+- `float` `.cooldown`
+- `bool` `.shipFriendlyFire`
+- `bool` `.hasSpecialPower`
+- `bool` `.hasTemporaryPower`
+- `int` `.jumpCooldown`
+- `int` `.disabledCooldown`
+- `float` `.initialCooldownFraction`
+- `int` `.onDeath`
+- `int` `.onHotkey`
+- `int` `.powerCharges`
+- `int` `.initialCharges`
+- `int` `.chargesPerJump`
+- `int` `.respawnCharges`
+- `int` `.disabledCharges`
+- `bool` `.hideCooldown`
+- `bool` `.hideCharges`
+- `bool` `.hideButton`
+- [`std::vector<PowerResourceDefinition*>`](#PowerResourceDefinition) `.powerResources`
+- `std::vector<std::string>` `.sounds`
+- `std::vector<std::string>` `.effectSounds`
+- `bool` `.soundsEnemy`
+- `bool` `.effectSoundsEnemy`
+- [`TextString`](#TextString) `.buttonLabel`
+- [`GL_Color`](#GL_Color) `.cooldownColor`
+- [`TextString`](#TextString) `.tooltip`
+- `std::string` `.effectAnim`
+- `std::string` `.effectPostAnim`
+- [`ActivatedPowerRequirements`](#ActivatedPowerRequirements) `.playerReq`
+- [`ActivatedPowerRequirements`](#ActivatedPowerRequirements) `.enemyReq`
+- [`ActivatedPowerRequirements*`](#ActivatedPowerRequirements) `.chargeReq`
+- `bool` `.win`
+- `float` `.crewHealth`
+- `float` `.enemyHealth`
+- `float` `.selfHealth`
+- `int` `.animFrame`
+- `bool` `.followCrew`
+- `bool` `.activateWhenReady`
+- `bool` `.activateReadyEnemies`
+- `std::string` `.transformRace`
+- [`std::vector<CrewSpawn*>`](#CrewSpawn) `.crewSpawns`
+- [`std::vector<StatBoostDefinition*>`](#StatBoostDefinition) `.statBoosts`
+- [`std::vector<StatBoostDefinition*>`](#StatBoostDefinition) `.roomStatBoosts`
+- `std::array<std::string,2>` `.event`
+- [`TemporaryPowerDefinition`](#TemporaryPowerDefinition) `.tempPower`
+
+## PowerResourceDefinition
+
+### Methods
+- `void :AssignIndex()`
+- `void :AssignName(std::string name)`
+- `void :AssignGroup(std::string name)`
+- `static` [`PowerResourceDefinition`](#PowerResourceDefinition) `.GetByName(std::string name)`
+- `static` [`PowerResourceDefinition`](#PowerResourceDefinition) `.AddNamedDefinition(std::string name, PowerResourceDefinition* copyDef);`
+
+### Fields
+- `std::string` `.name`
+   - **read-only**
+- `int` `.groupIndex`
+   - **read-only**
+- `int` `.sortOrder`
+- `float` `.cooldown`
+- `int` `.jumpCooldown`
+- `int` `.disabledCooldown`
+- `float` `.initialCooldownFraction`
+- `int` `.onDeath`
+- `int` `.powerCharges`
+- `int` `.initialCharges`
+- `int` `.chargesPerJump`
+- `int` `.respawnCharges`
+- `int` `.disabledCharges`
+- `bool` `.hideCooldown`
+- `bool` `.hideCharges`
+- `bool` `.showTemporaryBars`
+- `bool` `.showLinkedCooldowns`
+- `bool` `.showLinkedCharges`
+- [`GL_Color`](#GL_Color) `.cooldownColor`
+- [`ActivatedPowerRequirements*`](#ActivatedPowerRequirements) `.chargeReq`
+
+## ActivatedPowerRequirements
+
+### Fields
+- [`Type`](#Type) `.type`
+   - `int` `.PLAYER`
+   - `int` `.ENEMY`
+   - `int` `.CHARGE`
+   - `int` `.UNKNOWN`
```

Comment:
Not sure about this method of documenting an enum

Reply:
Frankly those enums thing were annoying to handle so if you have a better way I'm all ears